coverage_utils.py

```diff
@@ -17,17 +46,61 @@ def work(self, identifier):
         else:
             # Even if not, we can create our own LicensePool -- it's just
             # a stand-in and doesn't represent any actual licenses.
+            #
+            # This may happen because a migration script created work
+            # for this Identifier without going through
+            # IdentifierResolutionCoverageProvider.
             license_pool = self.license_pool(
                 identifier, data_source=DataSource.INTERNAL_PROCESSING
             )
             if not license_pool.licenses_owned:
                 license_pool.update_availability(1, 1, 0, 0)
 
+        # Making the dummy LicensePool open-access will ensure that
+        # when multiple collections have the same book, they'll
+        # all share a Work.
+        license_pool.open_access = True
+
+        # If the Identifier is already associated with a Work (because
+        # we went through this process for another LicensePool for the
+        # same identifier), we can reuse that Work and avoid a super()
+        # call, which will wastefully destroy the old Work and create
+        # an identical new one.
+        #
+        # Normally this isn't necessary because
+        # COVERAGE_COUNTS_FOR_EVERY_COLLECTION. But migration scripts
+        # may register seemingly redundant work to be done, and if
+        # that happens, we don't need to create a whole other Work
+        # -- we just need to recalculcate its presentation, which will
+        # happen in handle_success().
+        existing_work = identifier.work
+        if existing_work:
+            license_pool.work = existing_work
+            return existing_work
+
         return super(
             MetadataWranglerBibliographicCoverageProvider, self).work(
                 identifier, license_pool, even_if_no_title=True
             )
 
+    def handle_success(self, identifier):
+        """Try to create a new presentation-ready Work based on metadata
+        obtained during process_item().
+
+        If a Work already existed, recalculate its presentation to
+        incorporate the new metadata.
+        """
+        work = self.work(identifier)
+        if not isinstance(work, Work):
+            return work
+
+        if work.presentation_ready:
+            # This work was already presentation-ready, which means
+            # its presentation probably just changed and needs to be
+            # recalculated.
+            work.calculate_presentation()
```

Comment:
This may not technically be necessary because Metadata.apply will flag the work as needing its presentation recalculated. I like having it here because it lets me go to a URL like https://metadata.librarysimplified.org/lookup?urn=urn:isbn:9780062019622&resolve_now=true and see the new data immediately.